force-app/main/default/classes/LeadServiceTest.cls

```diff
@@ -1,15 +1,14 @@
 @isTest
-    private class LeadServiceTest {
```

Comment:
Why Class name is lowerCase? It should with Capital

LeadServiceTest

Reply:
Return your test as it was, it was correct. The only issue in LeadTriggerHandler